pkg/controller/podgc/gc_controller.go

```diff
@@ -124,6 +128,76 @@ func isPodTerminated(pod *v1.Pod) bool {
 	return false
 }
 
+// isPodTerminating returns true if the pod is terminating.
+func isPodTerminating(pod *v1.Pod) bool {
+	if phase := pod.Status.Phase; phase != v1.PodPending && phase != v1.PodRunning && phase != v1.PodUnknown {
+		return true
+	}
+	if pod.ObjectMeta.DeletionTimestamp != nil {
+		klog.Infof("garbage collecting pod %s that is terminating. Phase [%v]", pod.Name, pod.Status.Phase)
+		return true
+	}
+	return false
+}
+
+func (gcc *PodGCController) gcTerminating(pods []*v1.Pod) {
+	terminatedPods := []*v1.Pod{}
+	for _, pod := range pods {
+		if isPodTerminating(pod) {
+			node, err := gcc.nodeLister.Get(string(pod.Spec.NodeName))
+			if err != nil {
+				klog.Errorf("failed to get node:%s", err)
+				continue
+			}
+
+			// Add this pod to terminatedPods list only if the following conditions are met:
+			// 1. Node is not ready.
+			// 2. Node has `out-of-service` taint with `NoExecute` effect.
+			// 3. Pod does not tolerate the `out-of-service` taint with `NoExecute` effect.
+			taintToMatch := &v1.Taint{
+				Key:    taints.OutOfServiceTaint,
+				Effect: v1.TaintEffectNoExecute,
+			}
+			if !nodeutil.IsNodeReady(node) && taints.HasOutOfServiceTaintWithNoExecute(node.Spec.Taints) &&
+				!podToleratesTaint(taintToMatch, pod) {
+				terminatedPods = append(terminatedPods, pod)
+			}
```

Comment:
if inclement `deleteCount` here, we don't need https://github.com/kubernetes/kubernetes/pull/108486/files#diff-eaebfd5acb747dd3dd13eb812a745b0a886ba2a24f82e852772c45ac0d936f9fR168-R169 
```suggestion
				deleteCount++
			}
```

Reply:
I have used the following : 
```
deleteCount := len(terminatingPods)
	if deleteCount <= 0 {
		return
	}
```
I think that should suffice.